src/coreclr/tools/Common/TypeSystem/Interop/InteropStateManager.cs

```diff
@@ -493,7 +493,7 @@ protected override int GetValueHashCode(CalliMarshallingMethodThunk value)
                 return value.TargetSignature.GetHashCode();
             }
 
-            protected override bool CompareKeyToValue(MethodSignature key, CalliMarshallingMethodThunk value)
+            protected override bool CompareKeyToValue((MethodSignature Signature, bool UseRuntimeMarshalling) key, CalliMarshallingMethodThunk value)
             {
                 return key.Equals(value.TargetSignature);
```

Comment:
This is always `false`.

Besides fixing that, we need to incorporate the `UseRuntimeMarshalling` bit in the comparison and do the same in `CompareValueToValue` below.